### ISSUE TITLE
Not all variables in the JSON body get replaced

### ISSUE THREAD

**Issue by vladah** [a85/POSTMan-Chrome-Extension#5682]
Dear Postman Team,

I noticed that a certain POST request is failing as some of the variables don't get replaced:

For example I'm using nested variables to construct variable names to be able to set different environment variables for the same repeated requests:

{{variable1_{{variable2}}}}

The following JSON body is failing when submitted:

```
{
   "input":{
      "rule":{
         "rule_name":"{{rule_name}}",
         "src_endpoints":[
            {
               "object_uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "object_type":"SITE",
               "display_name":"{{site_name2}}",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "{{site_name2}}"
               ],
               "uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name2}}}}",
               "object_name":"{{site_name2}}"
            },
            {
               "object_uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "object_type":"SITE",
               "display_name":"{{site_name1}}",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "{{site_name1}}"
               ],
               "uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name1}}}}",
               "object_name":"{{site_name1}}"
            }
         ],
         "dst_endpoints":[
            {
               "object_uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "object_type":"SITE",
               "display_name":"{{site_name2}}",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "{{site_name2}}"
               ],
               "uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name2}}}}",
               "object_name":"{{site_name2}}"
            },
            {
               "object_uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "object_type":"SITE",
               "display_name":"{{site_name1}}",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "{{site_name1}}"
               ],
               "uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name1}}}}",
               "object_name":"{{site_name1}}"
            }
         ],
         "action":"PERMIT",
         "options":{
            "logging":{
               "enable":true
            }
         }
      },
      "policy_uuid":"{{AllSite_FW_Policy_{{tenant_name}}}}"
   }
}
```

From the failure message I can see that nested variables got replaced but not the enclosing (composite) variable:

{"status_code": "404", "error_tag": "No ID Found", "error_message": "The resource with id {{AllSite_FW_Policy_t3_g_psk_ns}} of type intent_policy does not exist", "error_app_message": "Unknown id: {{AllSite_FW_Policy_t3_g_psk_ns}}", "error_diag": "This error occurs when the resource with the specified id is not found", "error_code": "40027"}

On the other hand if I reduce the number of non nested variables like in the following request then everything is working fine. The following body although still has nested variables gets submitted OK:

```
{
   "input":{
      "rule":{
         "rule_name":"{{rule_name}}",
         "src_endpoints":[
            {
               "object_uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "object_type":"SITE",
               "display_name":"NFX18-19",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "NFX18-19"
               ],
               "uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name2}}}}",
               "object_name":"NFX18-19"
            },
            {
               "object_uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "object_type":"SITE",
               "display_name":"NFX17",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "NFX17"
               ],
               "uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name1}}}}",
               "object_name":"NFX17"
            }
         ],
         "dst_endpoints":[
            {
               "object_uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "object_type":"SITE",
               "display_name":"NFX18-19",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "NFX18-19"
               ],
               "uuid":"{{top_ser_uuid_{{site_name2}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name2}}}}",
               "object_name":"NFX18-19"
            },
            {
               "object_uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "object_type":"SITE",
               "display_name":"NFX17",
               "state":"PROVISIONED",
               "fq_name":[
                  "default-domain",
                  "{{tenant_name}}",
                  "NFX17"
               ],
               "uuid":"{{top_ser_uuid_{{site_name1}}}}",
               "uri":"/topology-service/site/{{top_ser_uuid_{{site_name1}}}}",
               "object_name":"NFX17"
            }
         ],
         "action":"PERMIT",
         "options":{
            "logging":{
               "enable":true
            }
         }
      },
      "policy_uuid":"{{AllSite_FW_Policy_{{tenant_name}}}}"
   }
}
```

The difference between the two bodies are only in NFX17 being {{site_name1}} and NFX18-19 being {{site_name2}}.

I wonder if there's some time out on replacing the variables or some other constraint that I'm hitting?

Thanks,

Vlad

Postman for Windows
Version 6.6.1
win32 10.0.17763 / x64

**Comment by abhijitkane** [comment]
Reproducible on Postman for OSX 6.6.0

**Comment by kamalaknn** [comment]
We've pushed the fix for this in Postman 6.7.1 😄

**Comment by vladah** [comment]
Thanks guys!

Vlad
